core/che-core-api-core/src/main/java/org/eclipse/che/api/core/Page.java

```diff
@@ -0,0 +1,312 @@
+        checkArgument(itemsBefore >= 0, "Required non-negative value of items before");
+        this.itemsBefore = itemsBefore;
+        checkArgument(pageSize > 0, "Required positive value of page size");
+        this.pageSize = pageSize;
+        checkArgument(totalCount >= 0, "Required non-negative value of total items");
+        this.totalCount = totalCount;
+    }
+
+    /** Returns true whether this page doesn't contain items, returns false if it does. */
+    public boolean isEmpty() {
+        return items.isEmpty();
+    }
+
+    /**
+     * Returns true when the current page has the next page,
+     * otherwise when the page is the last page false will be returned.
+     */
+    public boolean hasNextPage() {
+        return getNumber() != -1 && itemsBefore + pageSize < totalCount;
+    }
+
+    /**
+     * Returns true when this page has the previous page,
+     * otherwise when the page is the first page false will be returned.
+     */
+    public boolean hasPreviousPage() {
+        return getNumber() != -1 && itemsBefore != 0;
+    }
+
+    /**
+     * Returns an {@link Optional} describing reference to the next page,
+     * or the empty {@code Optional} when this page is the first one.
+     */
+    public Optional<PageRef> getNextPageRef() {
+        if (!hasNextPage()) {
+            return Optional.empty();
+        }
+        return Optional.of(new PageRef(itemsBefore + pageSize, pageSize));
+    }
+
+    /**
+     * Returns an {@link Optional} describing reference to the previous page,
+     * or the empty {@code Optional} when this page is the last one.
+     */
+    public Optional<PageRef> getPreviousPageRef() {
+        if (!hasPreviousPage()) {
+            return Optional.empty();
+        }
+        final long skipItems = itemsBefore <= pageSize ? 0 : itemsBefore - pageSize;
+        return Optional.of(new PageRef(skipItems, pageSize));
+    }
+
+    /** Returns the reference to the last page. */
+    public PageRef getLastPageRef() {
+        final long lastPageItems = totalCount % pageSize;
+        if (lastPageItems == 0) {
+            return new PageRef(totalCount <= pageSize ? 0 : totalCount - pageSize, pageSize);
+        }
+        return new PageRef(totalCount - lastPageItems, pageSize);
+    }
+
+    /** Returns the reference to the first page. */
+    public PageRef getFirstPageRef() {
+        return new PageRef(0, pageSize);
+    }
+
+    /**
+     * Returns the size of the current page.
+     *
+     * <p>Returned value is always positive and greater or equal
+     * to the value returned by the {@link #getItemsCount()} method.
+     */
+    public int getSize() {
+        return pageSize;
+    }
+
+    /**
+     * Returns page number starting from 1.
+     *
+     * <p>If the page is not regular page(it refers rather to the
+     * data window than to the certain page(e.g. skip=2, pageSize=4))
+     * then this method returns -1.
+     */
+    public long getNumber() {
```

Comment:
i think it is clearer in your examples because you used `someVar` name for variable, it looks okay when variable name somehow related to the _Page_ e.g.`page.getNumber()`.